src/main/config.cpp

```diff
@@ -294,12 +294,25 @@ IndexTypeSet &DBConfig::GetIndexTypes() {
 }
 
 void DBConfig::SetDefaultMaxMemory() {
-	auto memory = FileSystem::GetAvailableMemory();
-	if (!memory.IsValid()) {
-		options.maximum_memory = DBConfigOptions().maximum_memory;
+	const char *slurm_mem_per_node = getenv("SLURM_MEM_PER_NODE");
+	const char *slurm_mem_per_cpu = getenv("SLURM_MEM_PER_CPU");
+	const char *slurm_cpus_on_node = getenv("SLURM_CPUS_ON_NODE");
+	if (slurm_mem_per_node) {
+		options.maximum_memory = ParseMemoryLimitSlurm(slurm_mem_per_node) * 8 / 10;
 		return;
+	} else if (slurm_mem_per_cpu && slurm_cpus_on_node) {
+		idx_t mem_per_cpu = ParseMemoryLimitSlurm(slurm_mem_per_cpu);
+		idx_t num_threads = std::stoi(slurm_cpus_on_node);
```

Comment:
`std::stoi` throws when no conversion can be performed - can we use `TryCast::Operation<string_t, double>`  instead?